### DISCUSSION TITLE
Prefer a script in the user's config over an eval

### DISCUSSION THREAD

**Discussion by RivenSkaye** [JanDeDobbeleer/oh-my-posh#2544] [Ideas]
The current docs tell users to eval the output of running `oh-my-posh init [shell] [flags]` in their `.profile` or equivalent file.
It makes a lot more sense to me to save the command output to a script in standardized directories.
- `%APPDATA%\oh-my-posh` on Windows
- `~/.config/oh-my-posh` on Unix-likes
- Shell-specific paths for config and profile files

When writing my `oh-my-posh init <shell>` output to script files I already noticed env vars and special indicators like `~` don't get properly resolved everywhere, but resolving them before writing to `stdout` (or a file) is not an issue. A couple examples:
```sh
# Windows
> cmd /c "echo %APPDATA%"  # outputs something like "C:\Users\<username>\AppData\Roaming"
> powershell -c 'echo "$ENV:APPDATA"'
# Unix-like
$ bash -c "cd ~/.config ; pwd"
$ fish -c 'cd ~/.config; pwd' 
$ zsh -c 'cd ~/.config; pwd'
# Both
> nu -c '$nu.config-path | path dirname'
```
Then people would only have to add `source <confdir>/<scriptname>` (or equivalent for CMD and Powershell) in order to run their powerline command, which omp (if `oh-my-posh init` writes the file) could try to append to a sane default, or it could print the line to add to the relevant config file to stdout. The latter option would allow something along the lines of `oh-my-posh init bash --config ~/my_amazing.omp.json >> ~/.config/.bashrc`.
(except `~` in the init script doesn't resolve properly, so perhaps copy the file to `<confdir>/theme.omp.json`)

**Comment by JanDeDobbeleer** [comment]
@RivenSkaye you can already use `--print` for this. So `oh-my-posh init bash --print --config ~/my_amazing.omp.json >> ~/.config/.bashrc`

**Reply by RivenSkaye** [reply]
Yes, and output seems to be the same as when omitting it. Is there a reason to prefer `eval` over `source`, or is that some remnant of the past?

**Reply by JanDeDobbeleer** [reply]
@RivenSkaye it's 1 command vs multiple, and the default should be as accessible as possible. Plus, this way everything is always up to date. Initially it was exactly what you proposed, which is why you can still do that (if you want to be in control).
